Added TCs for VM Wizard Cloned Disk provision method

rhrazdil opened the pull request:
https://jira.coreos.com/browse/CNV-1041:
https://polarion.engineering.redhat.com/polarion/redirect/project/CNV/workitem?id=CNV-2446
https://polarion.engineering.redhat.com/polarion/redirect/project/CNV/workitem?id=CNV-2447

gouyang commented:
Better to review this after #403 is merged. :)

suomiy commented on frontend/integration-tests/tests/kubevirt/clone.vm.scenario.ts:
```diff
@@ -92,118 +88,115 @@ describe('Test clone VM.', () => {
+      await withResource(leakedResources, clonedVm.asResource(), async() => {
+        expect(searchYAML(`vm.kubevirt.io/name: ${vm.name}`, clonedVm.name, clonedVm.namespace, 'vm'))
+          .toBeTruthy('Cloned VM should have vm.kubevirt.io/name label.');
+        deleteResource(clonedVm.asResource());
```
is the reason for replacing `action('Delete')` with `deleteResource` only for separation of concern so that only one of our tests tests the `action('Delete')` ? Or is there any other reason?

rhrazdil replied:
Yes, I think it's not necessary to delete all VMs via UI action since this has a test case for itself. So it may be better to save some time and delete it via CLI.
But I'm open to change it back if you think it's more valuable to keep the whole user workflow.

suomiy replied:
I am okay with saving the time :) so long as it is tested at least in one place

suomiy commented on frontend/integration-tests/tests/kubevirt/clone.vm.scenario.ts:
```diff
@@ -92,118 +88,115 @@ describe('Test clone VM.', () => {
-      // Verify cloned disk dataVolumeTemplate is present in cloned VM manifest
-      expect(searchJSON('spec.dataVolumeTemplates[0].metadata.name', clonedVmDiskName, clonedVm.name, clonedVm.namespace, 'vm'))
-        .toBeTruthy('Cloned VM should have container image.');
+  describe('Test cloning with URL provision method', () => {
```
Can you please explain meaning of this test a bit? What does it test (i.e. I do not see any expect) ?

rhrazdil replied:
The diff is confusing here. This is just a beginning of a new describe block.
I have splitted the original describe block with all test cases into two separate blocks.
The reason is to have better managed resources - to have less resources created at a point in time.

suomiy replied:
ok

rhrazdil commented:
@suomiy Ready for review.
- I have additionally fixed tests for migration that originally didn't tolerate that the UI sometimes displays a new node name with some delay.
- added random names to tests with templates
- added await keywords in vm.overview tests

suomiy commented on frontend/integration-tests/tests/kubevirt/utils/utils.ts:
```diff
@@ -117,6 +117,22 @@ export async function logIn() {
+  const getRandByte = () => {
+    let byte: string;
+    do {
+      byte = Math.random().toString(36).replace(/[.g-z]/g, '').substr(2, 2);
```
we just need a random hex number right? Why not just `Math.random().toString(16)` ?

rhrazdil replied:
Yes, thanks. Updated

suomiy commented on frontend/integration-tests/tests/kubevirt/vm.wizard.scenario.ts:
```diff
@@ -45,30 +67,58 @@ describe('Kubevirt create VM using wizard', () => {
+        // Verify that DV of VM created with Cloned disk method points to correct PVC
+        const dvResource = getResourceObject(`${vm1.name}-${testDataVolume.metadata.name}-clone`, vm1.namespace, 'dv');
+        const pvcSource = _.get(dvResource, 'spec.source.pvc', {});
+        expect(pvcSource).toEqual({name: testDataVolume.metadata.name, namespace: testDataVolume.metadata.namespace});
```
Am I correct that we are trying to clone both VMs? Shouldn't we also check the 2nd VM here?

suomiy replied:
Or what is the reasoning behind having two VMs in this TC?

rhrazdil replied:
Tomas suggested to use one dataVolume for creating 2 VMs to ensure multiple VMs can be creted from single source.
So I decided to verify content of the first one since both VMs are created in the same way.

suomiy replied:
ok
